### ISSUE TITLE
508-defect-2 [SCREENREADER, COGNITION]: Rated disability headings should be logically structured and ordered

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#4784]
# [SCREENREADER, COGNITION]: Rated disability headings should be logically structured and ordered

## Description

When the screen reader reads individual disabilities, there is no context to understand which ones are included in the disability rating. Especially for those with cognitive challenges, it is likely to be very difficult to remember which items were service-connected relative to those that were not, and how it all goes together to inform their rating. 

**Josh's findings:**
- There are counting issues associated with description lists caught on Firefox/NVDA, VoiceOver/Safari,[ the previous a11y specialist recommended switching to an ordered list](https://github.com/department-of-veterans-affairs/va.gov-team/issues/4637#issuecomment-738500392). 
- There are also inherent issues with description lists on mobile screen readers (see video) which causes these lists to be extraordinarily verbose and difficult to parse through
- The need to use the definition element `dfn` was a bandaid fix to poor content writing to make it clear to screen reader users that "Service-connected disability?" is tied to the binary response "Yes/No"
- The recommendation to use headings to "sort" was likely made to simplify content copy and code

https://user-images.githubusercontent.com/14154792/178301111-fbd79865-0d2c-42b8-87d1-7ddc4415ea9b.MP4

You'll notice in this video that the use of `dt` announces the percentage as "60, term, Percent, term" among other oddities.

## Point of Contact

Josh Kim @joshkimux 

## Acceptance Criteria

As a screenreader user, I want to understand the how the individual disabilities comprise my disability rating.

## Steps to Recreate

1. Enter `https://staging.va.gov/disability/view-disability-rating/rating/` in browser
2. Start screenreading device 
3. Navigate to the content area where individual disabilities are listed
4. Imagine hearing this for the first time without seeing the screen
5. Verify that the mix of service-connected and not service-connected creates a confusing experience

## Questions with answers

> How does the data get to the page? (formatting)

[According to the product folder, the feature uses the vets-api via EVSS](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/disability/rated-disabilities#requirements-and-constraints); 

> Is service-connected vs. not the main differentiator?

According to Jesse "Service connected vs not is the main differentiator we use on the page." But it's still not a clear cut answer.

[From the discovery research:](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/disability/rated-disabilities#discovery-takeaways) "Veterans are well-aware of the general use of VA disability ratings, including the service-connected and non-service-connected aspects."

> Can it be sorted?

[The product folder states an assumption](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/disability/rated-disabilities#assumptions) that "Rated Disabilities will be lightly used by Veterans (who have instant recall of their Total Combined Disabilities Rating (TCDR))" which may imply that implementing a sort isn't a priority if this is a lightly used feature.

This tracks, and I think the original intent/recommendation was _not_ to implement a sorting function, but just a reshuffling of the heading structure and IA of the page to be easier to read.

## Possible Fixes (optional)

@skylerschain would like your take here on if this needs more discovery work (based on the questions above), or if we can move forward with the fix below:

A simple (potential) solution to this, would be to simply split apart service connected and non-service connected disabilities into their own separate sub-sections.

**For example:**

h1: Your VA disability ratings
- h2: XX% combined disability rating
- h2: Service connected
  - h3: 10% Allergies due to hearing loss
  - h3: 20% Hearing loss
- h2: Non service connected 
  - h3: Diabetes
  - h3: Gout

## WCAG or Vendor Guidance (optional)

[Cognitive, content linearization](https://webaim.org/techniques/screenreader/#linearization)

## Acceptance criteria
- [ ] Description lists are _not_ used for this pattern
- [ ] The experience is non-verbose to screen reader users
- [ ] Design approach does not deviate visually from current designs except what is necessary to add new header functionality that works with screen readers.

**Comment by jason-gcio** [comment]
![ratings](https://user-images.githubusercontent.com/53826255/124160772-9cfac200-da6a-11eb-86cd-abd0ebe5e00c.JPG)

**Comment by jason-gcio** [comment]
Other feedback included sorting the individual ratings to be descending (when "unmixed")

**Comment by jason-gcio** [comment]
This can go on back burner for now but is a great candidate for A/B testing

**Comment by saderagsdale** [comment]
Open questions:

1. How does the data get to the page? (formatting)
2. Is service-connected vs. not the main differentiator?
3. Can it be sorted? 
4. Best sorting method? (alpha, percentage, etc.)

Next steps: Determine if it's a blocker or not by talking to @joshkimux @jsokohl

**Comment by sortizsh** [comment]
see full conversation https://dsva.slack.com/archives/C5AGLBNRK/p1657216351709479?thread_ts=1657200078.524169&cid=C5AGLBNRK

**Comment by joshkimux** [comment]
I dug into this ticket a little bit more and picked up on the following:
- There are counting issues associated with description lists caught on Firefox/NVDA, VoiceOver/Safari,[ the previous a11y specialist recommended switching to an ordered list](https://github.com/department-of-veterans-affairs/va.gov-team/issues/4637#issuecomment-738500392). 
- There are also inherent issues with description lists on mobile screen readers (see video) which causes these lists to be extraordinarily verbose and difficult to parse through
- The need to use the definition element `dfn` was a bandaid fix to poor content writing to make it clear to screen reader users that "Service-connected disability?" is tied to the binary response "Yes/No"
- The recommendation to use headings to "sort" was likely made to simplify content copy and code

Will be looking to talk to designers and @AngelaFowler82 about simpler methods to fix these issues. I think we can reach a relatively easy decision 😄 

[I've made a codepen to help test as well](https://codepen.io/joshkimux/pen/abYZjem)

**Comment by joshkimux** [comment]
[I've created a mural documenting this ticket in more detail](https://app.mural.co/t/vsa8243/m/vsa8243/1657554097492/1beedff0c26515401a940590abdef3c913407896?sender=u2134c22982ad9c7b41798011).

**Comment by joshkimux** [comment]
@saderagsdale is there someone new we can assign this to by any chance?

**Comment by saderagsdale** [comment]
@joshkimux Yep yep - _view rated disabilities_ is a squad 1 item, so talk to Jacob and Skyler about prioritizing this one.

**Comment by skylerschain** [comment]
@joshkimux I'm not sure I understand your recommendation above just from reading this, but we can also add this one to the Design backlog and then maybe sync up to discuss the solution.

**Comment by joshkimux** [comment]
@skylerschain sweet. I've added this to the design backlog. I was hoping to mock out an example, but couldn't seem to get stuff to load on staging. I'll try again once we revisit this ❤️

**Comment by jacobworrell** [comment]
@joshkimux I have questions as to whether we're still the team to own this. This may or may not belong to the disability experience team. I'm unsure.

**Comment by joshkimux** [comment]
@jacobworrell I can check for you! I'll add this to my list to look into

**Comment by joshkimux** [comment]
@jacobworrell , this now belongs to the benefits management team. I'm not sure who their product owner is, but can do some digging.

**Comment by jacobworrell** [comment]
5 points when I get Zenhub working.

**Comment by jerekshoe** [comment]
@skylerschain @uxgary. Since we are going to be changing a lot of the code anyways, I was thinking it may make sense to consider using the `va-card` component so that we can align more with the new CST cards. Attached are some screenshots of what that could look like, vs the original

**Comment by jerekshoe** [comment]
Current design
![desktop-original](https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/a3ab252f-ba41-47d1-80a2-5ef55a55a7e3)
![mobile-original](https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/8572fba2-02af-4797-a381-2544f81ad5d3)

**Comment by jerekshoe** [comment]
With `va-card` web component
![desktop-va-card](https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/905e8a75-11c2-4589-a6f6-ccdd96810054)
![mobile-va-card](https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/fabf8f8b-8a45-4026-addd-d49a78a2a4d3)

**Comment by uxgary** [comment]
Summary of team conversation in slack, Currently the team doesn't get any data back that would allow us show if a veteran is permanently and totally disabled. Additionally, we don't have a way to get the max rating percentage for each condition nor do we get information back to specify if each condition is **static** or **non-static**. We will likely rely on the collaboration of other teams to implement those features. 

A reddit link was shared for [P&T (permanent and total) disability](https://www.reddit.com/r/VeteransBenefits/wiki/pt/)

**Comment by uxgary** [comment]
**Update**: I've created a [content intake ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/66049) for review.

**Comment by jerekshoe** [comment]
**Update (10/23):** I have completed an initial pass of the work needed and am trying to validate against [the sketch file](https://sketch.com/s/6186e45f-d6cb-4c59-aa84-60dbe4482be4) that @uxgary provided me with. I also found some other code improvements that I want to include in this work, including removing our usage of the `va-breadcrumbs` web component in favor of specifying the breadcrumbs we want in the `content-build` registry. I have [a PR in draft](https://github.com/department-of-veterans-affairs/vets-website/pull/26326) currently and am hoping to get that ready for review tomorrow (10/24). I need to capture some screenshots as there some visual differences and add them to the description.

**Comment by davidakennedy** [comment]
I wanted to drop in and comment to say that I've caught up on this ticket. The design direction looks like an excellent improvement from an accessibility/design standpoint. Excited to give it a test when it's ready.

**Comment by jerekshoe** [comment]
@davidakennedy Just wanted to get your thoughts on the current heading structure and see if there are any adjustments needed as we are manipulating font sizes quite a bit on this page to make the headers looks like they are levels they aren't
<img width="1046" alt="Screenshot 2023-10-25 at 2 27 17 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/2c071ede-f859-48db-a605-b62e1760d33d">
<img width="311" alt="Screenshot 2023-10-25 at 2 29 30 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/d94a65f3-47e5-450a-91bc-bf939320aa8a">

**Comment by jerekshoe** [comment]
**Note:** The new headings `Service-connected ratings` and `Conditions VA determined aren't service-connected` are both slated to be `h3`s

**Comment by davidakennedy** [comment]
@jerekshoe This looks solid. The one tweak I would make is changing the "Need help" `h3` to a `h2`. Because it isn't a subtopic of "Your individual ratings" but stands on its own and is more broad. The visual apparance can remain the same if that's the visual hierarchy from @uxgary.

How are you thinking about markup for the content inside the cards?
